Corrode currently has some bugs in translating loops. This patch
manually cleans up mis-translated loops until the underlying causes get
fixed.

Index: b/src/hash.rs
===================================================================
--- a/src/hash.rs
+++ b/src/hash.rs
@@ -318,7 +318,7 @@ pub unsafe extern fn insert_before(
         'loop4: loop {
             if q != (*list).hasharray[hashval as (usize)] {
                 if streq((*p).key as (*const u8),(*q).key as (*const u8)) {
-                    break 'loop4;
+                    return -1i32;
                 } else {
                     q = (*q).hashnext as (*mut hashnode);
                     continue 'loop4;
@@ -329,9 +329,9 @@ pub unsafe extern fn insert_before(
                 (*p).hashnext = q as (*mut hashnode);
                 (*(*p).hashprev).hashnext = p as (*mut hashnode);
                 (*q).hashprev = p as (*mut hashnode);
+                break 'loop4;
             }
         }
-        return -1i32;
     }
     (*p).next = marker as (*mut hashnode);
     (*p).prev = (*marker).prev;
